yellow-paper/docs/cryptography/performance-targets.md

```diff
@@ -0,0 +1,121 @@
+# Honk targets and win conditions
+
+## Introduction & context
+
+Aztec's cryptography tech stack and its associated implementation is an open-ended project with potential for many enhancements, optimisations and scope-creep.
+
+This document is designed to definitively answer the following questions:
+
+1. What are the metrics we care about when measuring our cryptography components?
+3. What are minimum satisfiable values for these metrics?
+4. What are the aspirational values for these metrics?
+
+# Important Metrics
+
+The following is a list of the relevant properties that affect the performance of the Aztec network:
+
+* Size of a user transaction (in kb)
+* Time to generate a user transaction proof
+* Memory required to generate a user transaction proof
+* Time to generate an Aztec Virtual Machine proof
+* Memory required to generate an Aztec Virtual Machine proof
+* Time to compute a 2x2 rollup proof
+* Memory required to compute a 2x2 rollup proof
+
+<!-- We can break these properties down into metrics linked to specitic cryptographic components:
+
+* Size of Goblin Plonk proofs
+* Size of Honk proofs
+* Honk prover time
+* Goblin Plonk prover time
+* Protogalaxy recursion -->
+
+ "MVP" = minimum standards that we can go to main-net with.
+
+| metric | how to measure | MVP | ideal |
+| --- | --- | --- | --- |
+| proof size | total size of a user tx incl. goblin plonk proofs | 32kb | 8kb |
+| prover time | 8 iterations of protogalaxy with 2^17 circuits (web browser) | 1 min | 10 seconds |
+| verifier time | how long does it take the verifier to check a proof (incl. grumpkin IPA MSMs) | 25ms | 1ms |
+| memory consumption | fold 2^17 circuits into an accumulator an arbitrary number of times | 2gb | 512mb |
+| size of the kernel circuit | number of gates | 2^17 | 2^15 |
+| Aztec Virtual Machine prover time | 1 million VM step circuit | 30 seconds | 5 seconds |
+| Aztec Virtual Machine memory consumption | 1 million VM step circuit | 128Gb | 16Gb |
+
+### Proof size
+
+At a tx throughput of 1,024 tx per second, each Aztec node (not sequencer/prover, just a regular node that is sending transactions) needs to download `1024*proof_size` bytes of data to keep track of the mempool. 32kb proofs = 32MB per second which is on the threshold of practical outside of a datacenter. 8MB per second is much more practical.
+
+At launch, the throughput won't be this high, but ideally we architect Aztec such that the network can scale effectively by throwing hardware+resources at bottlenecks (easy to do in a post-decentralisation), without protocol, architecture or tech upgrades ( harder post-decentralisation).
+
+### Prover time
+
+The critical UX factor. 1 minute per user tx is on the cusp of workable. The faster we can get this the better our network's value proposition for users and developers
```

Comment:
Perhaps we need to specify a kind of tx, since tx complexity will vary wildly. E.g. a simple transfer mimicking zcash audit functionality? Or a simple Aztec Connect style defi interaction (step 1 and step 2)?